src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4363,29 +4363,34 @@ def __init__(self, cmd, linux=False, windows=False):
         self.KEYS = FUNCTIONS_STACKS_API_KEYS()
         super().__init__(cmd, linux=linux, windows=windows)
 
-    def validate_end_of_life_date(self, runtime, version):
+    def validate_end_of_life_date(self, runtime, version, linux):
         from dateutil.relativedelta import relativedelta
         # we would not be able to validate for a custom runtime
         if runtime == 'custom':
             return
 
         today = datetime.datetime.now(datetime.timezone.utc)
         six_months = today + relativedelta(months=+6)
-        runtimes = [r for r in self.stacks if runtime == r.name]
+        runtimes = [r for r in self.stacks if r.linux == linux and runtime == r.name]
```

Comment:
> r.linux == linux

 This code looks a bit confusing

Reply:
For context, we support different runtimes depending on whether the customer wants to run their function app on windows or linux. So this linux param is a boolean to indicate whether the customer wants a linux function app.

Reply:
Got it, thanks